cardano-launcher/src/Cardano/Shell/Launcher.hs

```diff
@@ -74,6 +79,29 @@ data WalletMode
     | WalletModeSafe
     deriving (Eq, Show)
 
+instance Arbitrary WalletMode where
+  arbitrary = elements [ WalletModeSafe, WalletModeNormal ]
+
+instance FromJSON WalletMode where
+  parseJSON (Bool True) = pure WalletModeSafe
+  parseJSON (String "safe") = pure WalletModeSafe
+  parseJSON (String "normal") = pure WalletModeNormal
+  parseJSON _ = pure WalletModeNormal
+
```

Comment:
It should have a single representation, not map 5 types into this one?
If you want to have a `String`, use a `String`, otherwise use a `Bool`.
Don't mix and match, no need for that.

Reply:
the `Bool` stuff is backwards compatibility with the old launcher and `safemode.yaml`